src/IO/JSON/JSONIOHandlerImpl.cpp

```diff
@@ -54,9 +58,82 @@ namespace openPMD
             throw std::runtime_error((TEXT));                                  \
     }
 
-JSONIOHandlerImpl::JSONIOHandlerImpl(AbstractIOHandler *handler)
+namespace
+{
+    struct DefaultValue
+    {
+        template <typename T>
+        static nlohmann::json call()
+        {
+            if constexpr (auxiliary::IsComplex_v<T>)
+            {
+                return typename T::value_type{};
+            }
+            else
+            {
+                return T{};
+            }
+#if defined(__INTEL_COMPILER)
+/*
+ * ICPC has trouble with if constexpr, thinking that return statements are
```

Comment:
I think this is likely with EDG frontents in general and might also affect nvcc. I remember we saw this there as well and reported it at some point (should be fixed in newer >CUDA 12 versions).

Reply:
So, we should change the macro to something more precise?
If yes, do you know what would be the right check?